common/fs/src/cache/tailed_file.rs

```diff
@@ -56,8 +59,9 @@ fn read_until_internal<R: AsyncBufRead + ?Sized>(
         if done || used == 0 {
             if used == 0 {
                 Metrics::fs().increment_partial_reads();
+                return Poll::Ready(None);
```

Comment:
I think there's no need in adding a new set of possible states with `Option<T>`. Using `Poll::Ready(Ok(0))` should work.